File: fc-chat/demo/part3/ch3/05-1-message-system-add-channel/src/main/java/net/prostars/messagesystem/dto/websocket/outbound/MessageNotification.java
package net.prostars.messagesystem.dto.websocket.outbound;

import net.prostars.messagesystem.constant.MessageType;
import net.prostars.messagesystem.dto.domain.ChannelId;

public class MessageNotification extends BaseMessage {

  private final ChannelId channelId;
  private final String username;
  private final String content;

  public MessageNotification(ChannelId channelId, String username, String content) {
    super(MessageType.NOTIFY_MESSAGE);
    this.channelId = channelId;
    this.username = username;
    this.content = content;
  }

  public ChannelId getChannelId() {
    return channelId;
  }

  public String getUsername() {
    return username;
  }

  public String getContent() {
    return content;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/websocket/outbound/MessageNotification.kt
package com.yscorp.withpush.messagesystem.dto.websocket.outbound

import com.yscorp.withpush.messagesystem.constant.MessageType
import com.yscorp.withpush.messagesystem.dto.domain.ChannelId

class MessageNotification(val channelId: ChannelId, val username: String, val content: String) :
    BaseMessage(MessageType.NOTIFY_MESSAGE) {
}
